### ISSUE TITLE
PostgreSQL system administration functions fail due to no FROM clause

### ISSUE THREAD

**Issue by KnutVinsand** [psycopg/psycopg2#1767]
PostgreSQL has a series of built in administrative functions that you can see [in this URL](https://www.postgresql.org/docs/9.4/functions-admin.html), and as you can see in this sample, when you run pg_relation_size('TABLENAME'), you actually make no use of the FROM clause:

![image](https://github.com/user-attachments/assets/3ce39014-28a2-4228-b277-17b579adea0f)

However, when I try to query our database through psycopg2 and run this command, I am met with this error:

`psycopg2.errors.UndefinedTable: missing FROM-clause entry for table "SOMETABLE"
LINE 1: select pg_relation_size(SOMETABLE...
`

Normally, an SQL query is supposed to have a FROM statement, sure, but as you can see in the previously linked URL, and in the attached screenshot, these administrative functions run just fine without any FROM statements, and in fact, I cannot find a way to even get them to work *with* a FROM statement.

It's possible there is another way to make this work with psycopg2, but if there is, I'm not aware of.

If any additional information is needed, I'm naturally willing to help.

**Comment by dvarrazzo** [comment]
It seems that you are missing the quotes. What are you writing in psycopg?

**Comment by KnutVinsand** [comment]
Ah great, the error is mine then for trusting the error message and not checking my own code! Apologies, ticket closed!

**Comment by KnutVinsand** [comment]
As to answer your other question, I'm essentially making my own Pandas-esque module for data operations, except I was somewhat annoyed with having to use our internal tools to fetch the data that I needed because they were rather slow, so I created a direct connection to our database to allow me to query the tables I'd need without manually having to download it and move it to any specific folder for future use. I wanted to use the admin tools to give the user an estimate of how long it would take to download the query since, as far as I can tell from the testing I did yesterday, there was no good way of making a download progress tracker, so being able to present the user with the size of the download they were requesting would allow me to at least make an estimate. 

If you know of a solution, I'm all ears!
